BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,458 @@
+  var placeholder: String
+  @Binding var item: NetworkInputItem
+  
+  var body: some View {
+    VStack(alignment: .leading) {
+      TextField(placeholder, text: $item.input)
+        .autocapitalization(.none)
+        .disableAutocorrection(true)
+      if let error = item.error {
+        HStack(alignment: .firstTextBaseline, spacing: 4) {
+          Image(systemName: "exclamationmark.circle.fill")
+          Text(error)
+            .fixedSize(horizontal: false, vertical: true)
+              .animation(nil, value: error)
+        }
+        .accessibilityElement(children: .combine)
+        .transition(
+          .asymmetric(
+            insertion: .opacity.animation(.default),
+            removal: .identity
+          )
+        )
+        .font(.footnote)
+        .foregroundColor(Color(.braveErrorLabel))
+      }
+    }
+  }
+}
+
+class CustomNetworkModel: ObservableObject {
+  @Published var networkId = NetworkInputItem(input: "") {
+    didSet {
+      if networkId.input != oldValue.input {
+        if let intValue = Int(networkId.input), intValue > 0 {
+          networkId.error = nil
+        } else {
+          networkId.error = Strings.Wallet.customNetworkChainIdErrMsg
+        }
+      }
+    }
+  }
+  @Published var networkName = NetworkInputItem(input: "") {
+    didSet {
+      if networkName.input != oldValue.input {
+        if networkId.input.isEmpty {
+          networkName.error = Strings.Wallet.customNetworkEmptyErrMsg
+        } else {
+          networkName.error = nil
+        }
+      }
+    }
+  }
+  @Published var networkSymbolName = NetworkInputItem(input: "") {
+    didSet {
+      if networkSymbolName.input != oldValue.input {
+        if networkSymbolName.input.isEmpty {
+          networkSymbolName.error = Strings.Wallet.customNetworkEmptyErrMsg
+        } else {
+          networkSymbolName.error = nil
+        }
+      }
+    }
+  }
+  @Published var networkSymbol = NetworkInputItem(input: "") {
+    didSet {
+      if networkSymbol.input != oldValue.input {
+        if networkSymbol.input.isEmpty {
+          networkSymbol.error = Strings.Wallet.customNetworkEmptyErrMsg
+        } else {
+          networkSymbol.error = nil
+        }
+      }
+    }
+  }
+  @Published var networkDecimals = NetworkInputItem(input: "") {
+    didSet {
+      if networkDecimals.input != oldValue.input {
+        if networkDecimals.input.isEmpty {
+          networkDecimals.error = Strings.Wallet.customNetworkEmptyErrMsg
+        } else if let intValue = Int(networkDecimals.input), intValue > 0 {
+          networkDecimals.error = nil
+        } else {
+          networkDecimals.error = Strings.Wallet.customNetworkCurrencyDecimalErrMsg
+        }
+      }
+    }
+  }
+  
+  @Published var rpcUrls: [NetworkInputItem] = [NetworkInputItem(input: "")] {
+    didSet {
+      // we only care the set on each item's `input`
+      if rpcUrls.reduce("", { $0 + $1.input }) != oldValue.reduce("", { $0 + $1.input }) {
```

Comment:
I would say ditch this equality check as well I think

Reply:
same reason for this list of urls. every changes (either input or error) in any item will trigger didSet of this entire list.